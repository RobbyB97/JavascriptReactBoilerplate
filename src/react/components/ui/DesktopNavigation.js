/* Packages */
import React from 'react';
import {NavLink} from 'react-router-dom';
import {connect} from 'react-redux';


/* Component */
export class DesktopNavigation extends React.Component {
    constructor(props) {
        super(props);
    };

    render() {
        return (
            <section id="DesktopNavigation">
                <ul className="DesktopNavigation__list">
                    {this.props.internal_links &&
                        this.props.internal_links.map((link) => (
                            <li key={link.text}>
                                <a href={link.href}>
                                    {link.text}
                                </a>
                            </li>
                        ))
                    }
                </ul>
            </section>
        );
    };
};


export default DesktopNavigation;