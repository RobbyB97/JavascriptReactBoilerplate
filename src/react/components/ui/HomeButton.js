/* Packages */
import React from 'react';
import {NavLink} from 'react-router-dom';


/* Component */
export class HomeButton extends React.Component {
    constructor(props) {
        super(props);
    };


    render() {
        return (
            <section 
                id="HomeButton"    
            >
                <NavLink 
                    to="/"
                    className="HomeButton__logo"
                >
                    <img src="/dist/images/icons/ReactIconWhite.png" />
                </NavLink>
            </section>
        );
    };
};


export default HomeButton;