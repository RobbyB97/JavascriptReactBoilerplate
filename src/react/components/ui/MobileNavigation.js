/* Packages */
import React from 'react';
import {NavLink} from 'react-router-dom';
import {connect} from 'react-redux';


/* Component */
export class MobileNavigation extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            navMenu_Mobile: false
        };
    };

    
    navMenu_Mobile__Toggle = () => {
        this.props.navMenu_Mobile__Toggle();
    };


    render() {
        return (
            <section id="MobileNavigation">
                <aside 
                    data-active={this.state.navMenu_Mobile}
                    className="MobileNavigation__Menu"
                >
                    <ul className="MobileNavigation__list">
                        {this.props.internal_links &&
                            this.props.internal_links.map((link) => (
                                <li key={link.text}>
                                    <a href={link.href}>
                                        {link.text}
                                    </a>
                                </li>
                            ))
                        }
                    </ul>
                </aside>
            </section>
        );
    };
};


/* Connect to store */
const mapStateToProps = (state) => {
    return {
        navMenu_Mobile: state.ui.navMenu_Mobile
    };
};


const mapDispatchToProps = (dispatch) => ({
    navMenu_Mobile__Toggle: () => {
        dispatch(navMenu_Mobile__Toggle());
    }
});


export default MobileNavigation;