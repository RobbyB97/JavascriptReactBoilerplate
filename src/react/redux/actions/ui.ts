import { 
    MOBILE_NAV__TOGGLE, MOBILE_NAV__ON, MOBILE_NAV__OFF, UIActionTypes 
} from '../types/ui';


export function navMenu_Mobile__Toggle(): UIActionTypes {
    return {
        type: MOBILE_NAV__TOGGLE
    }
}

export function navMenu_Mobile__On(): UIActionTypes {
    return {
        type: MOBILE_NAV__ON
    }
}

export function navMenu_Mobile__Off(): UIActionTypes {
    return {
        type: MOBILE_NAV__OFF
    }
}