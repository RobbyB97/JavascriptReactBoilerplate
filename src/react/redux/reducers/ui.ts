import { 
    UIState, MOBILE_NAV__TOGGLE, MOBILE_NAV__ON, MOBILE_NAV__OFF, UIActionTypes 
} from '../types/ui';

const initialState: UIState = {
    navMenu_Mobile: false
};

export function uiReducer(
    state = initialState,
    action: UIActionTypes
): UIState {
    switch(action.type) {
        case MOBILE_NAV__TOGGLE:
            return {
                navMenu_Mobile: state.navMenu_Mobile ? false : true
            }

        case MOBILE_NAV__ON:
            return {
                navMenu_Mobile: true
            }

        case MOBILE_NAV__OFF:
            return {
                navMenu_Mobile: false
            }

        default:
            return state;
    }
}


/*
export default (state = initialState, action) => {
    switch(action.type) {
        case 'MOBILE_NAV__TOGGLE':
            return {
                navMenu_Mobile: state.navMenu_Mobile ? false : true
            };

        case 'MOBILE_NAV__ON':
            return {
                navMenu_Mobile: true
            };
    

        case 'MOBILE_NAV__OFF':
            return {
                navMenu_Mobile: false
            };

        default:
            return state;
    };
};
*/