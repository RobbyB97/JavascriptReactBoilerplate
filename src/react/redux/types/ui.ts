/* State types */
export interface UIState {
    navMenu_Mobile: boolean;
}


/* Actions */
export const MOBILE_NAV__TOGGLE = 'MOBILE_NAV__TOGGLE';
export const MOBILE_NAV__ON = 'MOBILE_NAV__ON';
export const MOBILE_NAV__OFF = 'MOBILE_NAV__OFF';

interface MobileNavToggleAction {
    type: typeof MOBILE_NAV__TOGGLE;
}

interface MobileNavOnAction {
    type: typeof MOBILE_NAV__ON;
}

interface MobileNavOffAction {
    type: typeof MOBILE_NAV__OFF;
}

export type UIActionTypes = MobileNavToggleAction | MobileNavOnAction | MobileNavOffAction;