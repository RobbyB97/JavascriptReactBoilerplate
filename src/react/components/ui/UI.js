/* Packages */
import React from 'react';
import {connect} from 'react-redux';


/* Components */
import HomeButton from './HomeButton';
import DesktopNavigation from './DesktopNavigation/DesktopNavigation';
import MobileNavigation from './MobileNavigation/MobileNavigation';


/* Component */
export class UI extends React.Component {
    constructor(props) {
        super(props);
    };

    render() {
        const internal_links = [
            {
                route: "/",
                text: "Home",
                sublinks: [{
                    text: "Home",
                    route: "/"
                }]
            }
        ]
        const external_links = [
            {
                text: "Meet the Dev",
                href: "https://bergers.dev",
                external: true
            }, {
                text: "View on GitHub",
                href: "https://github.com/RobbyB97/JavascriptReactBoilerplate",
                external: true,
                sublinks: [{
                    text: "Test 1",
                    href: "google.com",
                    external: true
                }, {
                    text: "Test 1",
                    href: "google.com",
                    external: true                    
                }]
            }
        ]

        return (
            <section id="UI">
                <HomeButton />

                <DesktopNavigation 
                    internal_links={internal_links}
                    external_links={external_links}
                />
                
                <MobileNavigation 
                    internal_links={internal_links}
                    external_links={external_links}                    
                />
            </section>
        );
    };
};


export default UI;