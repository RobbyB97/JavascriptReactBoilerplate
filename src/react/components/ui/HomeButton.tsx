import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';

import {navMenu_Mobile__Toggle} from '../../redux/actions/ui';


interface Props {
    navMenu_Mobile: boolean;
    navMenu_Mobile__Toggle: () => void;
}

interface State {

}


export class HomeButton extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props);
    };

    navMenu_Mobile__Toggle = () => {
        /* Enable scrolling */
        const html = document.getElementsByTagName('HTML')[0];
        html.setAttribute('data-Mobile_Nav', 'false');
        
        this.props.navMenu_Mobile__Toggle();
    };

    render() {
        return (
            <section 
                id="HomeButton"
                onClick={this.props.navMenu_Mobile ? this.navMenu_Mobile__Toggle : undefined}
            >
                <NavLink 
                    to="/"
                    className="HomeButton__logo"
                >
                    {this.props.navMenu_Mobile ?
                        /* Nav Menu HomeButton */
                        <img src="/dist/images/homeButton/HomeButtonMobileNav.png" alt="Home"/>
                        :

                        /* Default HomeButton */
                        <img src="/dist/images/homeButton/HomeButton.png" alt="Home"/>
                    }
                </NavLink>
            </section>
        );
    }
}


/* Connect to store */
const mapStateToProps = (state: any) => {
    return {
        navMenu_Mobile: state.ui.navMenu_Mobile
    };
};


const mapDispatchToProps = (dispatch: Dispatch<any>) => ({
    navMenu_Mobile__Toggle: () => {
        dispatch(navMenu_Mobile__Toggle());
    }
});


export default connect(mapStateToProps, mapDispatchToProps)(HomeButton);