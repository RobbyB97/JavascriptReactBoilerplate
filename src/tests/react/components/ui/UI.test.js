/* Packages */
import React from 'react';
import {shallow} from 'enzyme';


/* Component */
import {UI} from '../../../../react/components/ui/UI';


/* Test Config */
let component;
beforeEach(() => {
    component = shallow(<UI />);
});


/* Tests */
test('Render UI component', () => {
    expect(component).toMatchSnapshot();
});

