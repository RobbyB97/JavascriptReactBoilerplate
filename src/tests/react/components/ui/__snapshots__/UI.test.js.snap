// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render UI component 1`] = `
<section
  id="UI"
>
  <Connect(HomeButton) />
  <DesktopNavigation
    links={
      Array [
        Object {
          "external": true,
          "href": "https://bergers.dev",
          "text": "Meet the Dev",
        },
        Object {
          "external": true,
          "href": "https://github.com/RobbyB97/JavascriptReactBoilerplate",
          "text": "View on GitHub",
        },
      ]
    }
    navlinks={
      Array [
        Object {
          "route": "/",
          "text": "Home",
        },
      ]
    }
  />
  <Connect(MobileNavigation)
    links={
      Array [
        Object {
          "external": true,
          "href": "https://bergers.dev",
          "text": "Meet the Dev",
        },
        Object {
          "external": true,
          "href": "https://github.com/RobbyB97/JavascriptReactBoilerplate",
          "text": "View on GitHub",
        },
      ]
    }
    navlinks={
      Array [
        Object {
          "route": "/",
          "text": "Home",
        },
      ]
    }
  />
</section>
`;
